pylearn2/sandbox/nlp/models/tests/composite.yaml

```diff
@@ -0,0 +1,78 @@
+!obj:pylearn2.train.Train {
+    dataset: &train !obj:pylearn2.datasets.vector_spaces_dataset.VectorSpacesDataset {
+        data: !obj:pylearn2.sandbox.nlp.models.as_tuple { list_: [
```

Comment:
In that case, I'm pretty sure there is no particular reason. It may even already work, and the only change needed would be to update the docstring :)
If it does not, then changing the `__init__` method to set `self.data = tuple(data)` should be enough.